File: macos/QMK Toolbox/USB/Bootloader/SN32DFUDevice.swift
import Foundation

class SN32DFUDevice: BootloaderDevice {
    override init(usbDevice: USBDevice) {
        super.init(usbDevice: usbDevice)
        name = "SN32 DFU"
        type = .sn32Dfu
    }

    override func flash(_ mcu: String, file: String) {
        if productID == 0x7010 {
            runProcess("sonixflasher", args: ["-v", String(format: "%04X:%04X", vendorID, productID), "-o", "0x200", "-f", file])
        } else {
            runProcess("sonixflasher", args: ["-v", String(format: "%04X:%04X", vendorID, productID), "-f", file])
        }
    }
}

File: windows/QMK Toolbox/Usb/Bootloader/Sn32DfuDevice.cs
using System.IO;
using System.Threading.Tasks;

namespace QMK_Toolbox.Usb.Bootloader
{
    class Sn32DfuDevice : BootloaderDevice
    {
        public Sn32DfuDevice(UsbDevice d) : base(d)
        {
            Type = BootloaderType.Sn32Dfu;
            Name = "SN32 DFU";
            PreferredDriver = "HidUsb";
        }

        public async override Task Flash(string mcu, string file)
        {
            if (ProductId == 0x7010) // SN32F260
            {
                await RunProcessAsync("sonixflasher.exe", $"-v {VendorId:x4}:{ProductId:x4} -o 0x200 -f \"{file}\"");
                return;
            }

            await RunProcessAsync("sonixflasher.exe", $"-v {VendorId:x4}:{ProductId:x4} -f \"{file}\"");
        }
    }
}
